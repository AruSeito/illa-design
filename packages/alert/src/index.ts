export * from "./alert"
export * from "./interface"
