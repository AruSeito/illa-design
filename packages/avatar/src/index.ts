export * from "./avatar"
export * from "./interface"
export * from "./avatar-group"