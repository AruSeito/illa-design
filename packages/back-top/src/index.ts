export * from "./backtop"
export * from "./interface"
export * from "./style"
