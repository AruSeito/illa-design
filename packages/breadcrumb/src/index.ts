export * from "./interface"
export * from "./breadcrumb"
export * from "./breadcrumbItem"
