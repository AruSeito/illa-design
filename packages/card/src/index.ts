export * from "./card"
export * from "./interface"
export * from "./grid"
export * from "./meta"
