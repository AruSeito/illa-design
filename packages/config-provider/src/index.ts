export * from "./config-provider"
export * from "./interface"

// locale
export * from "./locale/en-US"
export * from "./locale/zh-CN"
export * from "./locale/def"
