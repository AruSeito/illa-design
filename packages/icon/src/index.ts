export * from "./icon"
export * from "./interface"
export * from "./create-icon"
// icons
export * from "./icons/close"
export * from "./icons/person"
export * from "./icons/image-default"
export * from "./icons/loading"
export * from "./icons/copy"
export * from "./icons/eye-on"
export * from "./icons/eye-off"