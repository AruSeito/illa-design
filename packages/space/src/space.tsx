/** @jsxImportSource @emotion/react */
import * as React from "react"
import { Children, forwardRef, Fragment } from "react"
import { SpaceAlign, SpaceDirection, SpaceProps, SpaceSize } from "./interface"
import { css, SerializedStyles } from "@emotion/react"
import { Divider, DividerDirection } from "@illa-design/divider"

function applyContainer(direction: SpaceDirection, align: SpaceAlign, wrap: boolean): SerializedStyles {
  let cssDirection: string
  switch (direction) {
    case "horizontal":
      cssDirection = "row"
      break
    case "vertical":
      cssDirection = "column"
      break
  }
  return css`
    display: inline-flex;
    vertical-align: middle;
    flex-direction: ${cssDirection};
    align-items: ${align};
    flex-wrap: ${wrap ? "wrap" : "nowrap"};
  `
}

function applyDividerSize(size: SpaceSize | SpaceSize[], isStart: boolean): SerializedStyles {
  let horSpace, verSpace: string

  if (typeof size == "string") {
    horSpace = verSpace = getSpaceSize(size, isStart)
  } else {
    if (size.length == 1) {
      horSpace = verSpace = getSpaceSize(size[0], isStart)
    } else if (size.length >= 2) {
      horSpace = getSpaceSize(size[0], isStart)
      verSpace = getSpaceSize(size[1], isStart)
    } else {
      horSpace = verSpace = getSpaceSize(size[0], isStart)
    }
  }
  return css`
    margin-right: ${horSpace};
    margin-bottom: ${verSpace};
  `
}

function getSpaceSize(size: SpaceSize, isStart: boolean): string {
  if (isStart) {
    return "0px"
  } else {
    switch (size) {
      case "mini":
        return "4px"
      case "small":
        return "8px"
      case "medium":
        return "16px"
      case "large":
        return "24px"
      default:
        return size
    }
  }
}

export const Space = forwardRef<HTMLDivElement, SpaceProps>((props, ref) => {

  const {
    size = "small",
    align = "center",
    direction = "horizontal",
    divider = false,
    wrap = false,
    ...otherProps
  } = props

  let childrenArray = Children.toArray(props.children)

  return <div css={applyContainer(direction, align, wrap)} ref={ref} {...otherProps}>
    {childrenArray.map((child, index) => {
      return <Fragment key={index}>
        {index != 0 && divider ? <Divider css={applyDividerSize(size, index == 0)}
                                          direction={direction == "horizontal" ? "vertical" as DividerDirection : "horizontal" as DividerDirection} /> : null}
        <div css={applyDividerSize(size, index == 0)}>
          {child}
        </div>
      </Fragment>
    })}
  </div>

})
