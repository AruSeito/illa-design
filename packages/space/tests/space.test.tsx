import * as React from "react"
import { render,screen } from "@testing-library/react"
import { Space } from "../src"
import "@testing-library/jest-dom"

test("Space renders with different size", () => {
  render(<div>
    <Space size="mini">
    <span>
      A
    </span>
      <span data-testid="test-space-mini">
      B
    </span>
    </Space>
    <Space size="small">
    <span>
      A
    </span>
      <span data-testid="test-space-small">
      B
    </span>
    </Space>
    <Space size="medium">
    <span>
      A
    </span>
      <span data-testid="test-space-medium">
      B
    </span>
    </Space>
    <Space size="large">
    <span>
      A
    </span>
      <span data-testid="test-space-large">
      B
    </span>
    </Space>
    <Space size="30px">
    <span>
      A
    </span>
      <span data-testid="test-space-custom">
      B
    </span>
    </Space>
  </div>)
  expect(screen.getByTestId("test-space-mini").parentNode).toHaveStyle({
    "margin-left": "4px",
  })
  expect(screen.getByTestId("test-space-small").parentNode).toHaveStyle({
    "margin-left": "8px",
  })
  expect(screen.getByTestId("test-space-medium").parentNode).toHaveStyle({
    "margin-left": "16px",
  })
  expect(screen.getByTestId("test-space-large").parentNode).toHaveStyle({
    "margin-left": "24px",
  })
  expect(screen.getByTestId("test-space-custom").parentNode).toHaveStyle({
    "margin-left": "30px",
  })
})

test("Space renders with different direction", () => {
  render(<div>
    <Space direction="horizontal">
    <span>
      A
    </span>
      <span data-testid="test-space-mini">
      B
    </span>
    </Space>
    <Space direction="vertical">
    <span>
      A
    </span>
      <span data-testid="test-space-small">
      B
    </span>
    </Space>
  </div>)
  expect(screen.getByTestId("test-space-mini").parentNode).toHaveStyle("margin-left:8px")
  expect(screen.getByTestId("test-space-small").parentNode).toHaveStyle("margin-top:8px")
})

test("Space renders with different align", () => {
  const { asFragment } = render(
    <Space align="start">
    <span>
      A
    </span>
      <span>
      B
    </span>
    </Space>,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("Space renders with divider", () => {
  const { asFragment } = render(
    <Space divider={true}>
    <span>
      A
    </span>
      <span>
      B
    </span>
    </Space>,
  )
  expect(asFragment()).toMatchSnapshot()
})