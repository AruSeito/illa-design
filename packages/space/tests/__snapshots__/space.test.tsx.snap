// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space renders with different align 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-1 {
  margin-right: 0px;
  margin-bottom: 0px;
}

.emotion-2 {
  margin-right: 8px;
  margin-bottom: 8px;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span>
        A
      </span>
    </div>
    <div
      class="emotion-2"
    >
      <span>
        B
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Space renders with divider 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-1 {
  margin-right: 0px;
  margin-bottom: 0px;
}

.emotion-2 {
  margin-right: 8px;
  margin-bottom: 8px;
}

.emotion-3 {
  display: inline-flex;
  vertical-align: middle;
  border-width: 0 0 0 1px;
  border-image: initial;
  border-color: #ebebeb;
  border-style: solid;
  height: 1em;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span>
        A
      </span>
    </div>
    <div
      class="emotion-2 emotion-3"
    />
    <div
      class="emotion-2"
    >
      <span>
        B
      </span>
    </div>
  </div>
</DocumentFragment>
`;
